src/front/wgsl/mod.rs

```diff
@@ -1257,6 +1257,72 @@ impl Parser {
             }
         } else {
             match name {
+                "bitcast" => {
+                    let _ = lexer.next();
+                    lexer.expect_generic_paren('<')?;
+                    let ((ty, _access), type_span) = lexer.capture_span(|lexer| {
+                        self.parse_type_decl(lexer, None, ctx.types, ctx.constants)
+                    })?;
+                    lexer.expect_generic_paren('>')?;
+
+                    let expr = self.parse_general_expression(lexer, ctx.reborrow())?;
+
+                    // Create error that will be returned if type matching fails
+                    let err = Err(Error::BadTypeCast {
+                        from_type: format!("{:?}", ctx.resolve_type(expr)?),
+                        span: type_span,
+                        to_type: format!("{:?}", ctx.types[ty].inner),
+                    });
+
+                    match ctx.types[ty].inner {
+                        crate::TypeInner::Scalar { kind, .. } => match *ctx.resolve_type(expr)? {
+                            crate::TypeInner::Scalar { .. } => crate::Expression::As {
+                                expr,
+                                kind,
+                                convert: None,
+                            },
+                            _ => return err,
+                        },
+                        crate::TypeInner::Vector { size, kind, .. } => {
+                            match *ctx.resolve_type(expr)? {
+                                crate::TypeInner::Vector { .. } => {
+                                    let count = match size {
+                                        crate::VectorSize::Bi => 2,
+                                        crate::VectorSize::Tri => 3,
+                                        crate::VectorSize::Quad => 4,
+                                    };
+
+                                    // For vectors, cast each field individually
```

Comment:
we shouldn't need this either. Generally we are trying to make WGSL the textual form of our IR. So if WGSL supports this on vectors, we'll need to support it in the IR.

Reply:
After trying it, it actually looks like the IR is already fine with us casting vectors using the `As` expression. It seems like OpenGL needs a fix to properly cast vectors because `intBitsToFloat` doesn't support vectors, but that can be a separate issue/PR.